components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubEventSubscriberImpl.java

```diff
@@ -100,32 +102,56 @@ public void unsubscribe(List<String> channels, String eventProfileVersion, Strin
     }
 
     private void makeSubscriptionAPICall(String topic, String webSubHubBaseUrl, String operation, String callbackUrl,
-                                         String secret)
-            throws WebSubAdapterException {
+                                         String secret) throws WebSubAdapterException {
 
         ClientManager clientManager = WebSubHubAdapterDataHolder.getInstance().getClientManager();
-        HttpPost httpPost = clientManager.createHttpPost(webSubHubBaseUrl, null);
 
-        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8");
-
-        //TODO: add build url based on operation
-        List<BasicNameValuePair> params = new ArrayList<>();
-        params.add(new BasicNameValuePair(HUB_CALLBACK, callbackUrl));
-        params.add(new BasicNameValuePair(HUB_MODE, operation));
-        params.add(new BasicNameValuePair(HUB_TOPIC, topic));
-        if (secret != null) {
-            params.add(new BasicNameValuePair(HUB_SECRET, secret));
-        }
-
-        httpPost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
-
-        WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(httpPost);
-        final long requestStartTime = System.currentTimeMillis();
-
-        try (CloseableHttpResponse response = (CloseableHttpResponse) clientManager.execute(httpPost)) {
-            handleSubscriptionResponse(response, httpPost, topic, operation, requestStartTime);
-        } catch (IOException | WebSubAdapterException e) {
-            throw handleServerException(ERROR_SUBSCRIBING_TO_TOPIC, e);
+        int attempt = 0;
+        while (true) {
+            HttpPost httpPost = clientManager.createHttpPost(webSubHubBaseUrl, null);
+            httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8");
+
+            List<BasicNameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(HUB_CALLBACK, callbackUrl));
+            params.add(new BasicNameValuePair(HUB_MODE, operation));
+            params.add(new BasicNameValuePair(HUB_TOPIC, topic));
+            if (secret != null) {
+                params.add(new BasicNameValuePair(HUB_SECRET, secret));
+            }
+            httpPost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
+
+            WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(httpPost);
+            final long requestStartTime = System.currentTimeMillis();
+
+            try (CloseableHttpResponse response = (CloseableHttpResponse) clientManager.execute(httpPost)) {
+                int responseCode = response.getStatusLine().getStatusCode();
+                if (responseCode >= 500 && attempt < MAX_RETRIES) {
+                    attempt++;
+                    log.info("Retrying subscription API call, attempt " + attempt + " for topic: " + topic);
```

Comment:
This should be a debug and should have a diagnostic log ideally

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/18/commits/2c832d024b4ec63f738b137c01c3d87b2db08e3d